### ISSUE TITLE
Error "UnboundLocalError: local variable 'paths' referenced before assignment"

### ISSUE THREAD

**Issue by SebastienNin** [guillaumecharbonnier/mw-lib#30]
Hi @guillaumecharbonnier,
I installed Mw on a new server having snakemake 5.18.1 and python 3.10.8

When running an RNA-seq analysis, I got the following error.

```
InputFunctionException in line 78 of /data/TGML/mw-lib/src/snakemake/rules/bowtie2.smk:
UnboundLocalError: local variable 'paths' referenced before assignment                                        
Wildcards:                                                                                                                                                                                                                                                                         
tool=bowtie2/pe                                                                                                                                                                                                                                                                    
extra=                                                                                                                                   
index=Rnor6                                                                                                                                                                                                                                                                        
filler=sickle/pe_-t_sanger_-q_20/ln/alias/tgml/all_samples/fastq/S004812_1_E17-M1-1305-G
```

I didn't change anything except the server...
Do you have any idea of how I can find the source of the problem?

Sebastien

**Comment by guillaumecharbonnier** [comment]
The only difference I see is the updated python version. Could you check the python version in your working Sacapus env and then redeploy a snakemake version with the same version?
Could you create an sftp to share everything for me to reproduce on my workstation?

**Comment by SebastienNin** [comment]
Hi Guillaume, on sacapus, I used the version 3.10.5.
The problem must come from my rework of the Sequencing_summary. I changed many column name. I'll check again everything.

Here is the repo of mw-tgml (https://github.com/TGMLplatform/mw-tgml)
I'll create a sFTP and send you the Sequencing_summary I used with the corresponding input folder.

**Comment by SebastienNin** [comment]
I tried with python 3.10.5 and python 3.6 and the error appeared again

**Comment by guillaumecharbonnier** [comment]
I have looked at the code without testing it with your data (sftp transfert ongoing...).
I think the issue is just that there is no Rnor6 assembly defined here:
https://github.com/guillaumecharbonnier/mw-lib/blob/master/src/snakemake/tables/bowtie2_index_parts.tsv

I have never processed Rat genome with bowtie2 before. 
Feel free to write and merge the fix commit directly to master if this is actually the cause, and let me know otherwise.

**Comment by SebastienNin** [comment]
I found it this morning... I'm making some tests ans i'll commit it soon

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by SebastienNin** [comment]
Thanks for your help :)

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by SebastienNin** [comment]
Done and merged in commit [e43ac12](https://github.com/guillaumecharbonnier/mw-lib/commit/e43ac1277353810681c13467c5649951f97d3ca8)
